kf/nfconfig_test.go

```diff
@@ -417,7 +417,511 @@ func Test_getHostInterfaces(t *testing.T) {
+					XDPIngress: []*models.BPFProgram{},
+					TCEgress:   []*models.BPFProgram{},
+					TCIngress:  []*models.BPFProgram{},
+				},
+			},
+			wanterr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := &NFConfigs{
+				hostName:       tt.field.hostName,
+				IngressXDPBpfs: tt.field.ingressXDPBpfs,
+				IngressTCBpfs:  tt.field.ingressTCBpfs,
+				EgressTCBpfs:   tt.field.egressTCBpfs,
+				hostConfig:     tt.field.hostConfig,
+				processMon:     tt.field.processMon,
+				hostInterfaces: tt.field.hostInterfaces,
+				mu:             tt.field.mu,
+			}
+			err := cfg.AddProgramsOnInterface(tt.arg.iface, tt.arg.hostName, tt.arg.bpfProgs)
+			if (err != nil) != tt.wanterr {
+				t.Errorf("AddProgramsOnInterface: %v", err)
+			}
+		})
+	}
+}
+
+func TestAddeBPFPrograms(t *testing.T) {
+	type fields struct {
+		hostName       string
+		hostInterfaces map[string]bool
+		ingressXDPBpfs map[string]*list.List
+		ingressTCBpfs  map[string]*list.List
+		egressTCBpfs   map[string]*list.List
+		hostConfig     *config.Config
+		processMon     *pCheck
+		mu             *sync.Mutex
+		ifaces         map[string]string
+	}
+	tests := []struct {
+		name    string
+		field   fields
+		arg     []models.L3afBPFPrograms
+		wanterr bool
+	}{
+		{
+			name: "UnknowHostName",
+			field: fields{
+				hostName: "l3af-prod",
+				ifaces:   map[string]string{},
+				hostConfig: &config.Config{
+					L3afConfigStoreFileName: "/etc/l3afd/l3af-config.json",
+				},
+			},
+			arg: []models.L3afBPFPrograms{
+				{
+					HostName: "l3af-test",
+					Iface:    "enp0s3",
+					BpfPrograms: &models.BPFPrograms{
+						XDPIngress: []*models.BPFProgram{},
+						TCIngress:  []*models.BPFProgram{},
+						TCEgress:   []*models.BPFProgram{},
+					},
+				},
+			},
+			wanterr: true,
+		},
+		{
+			name: "NilInterface",
+			field: fields{
+				hostName: "l3af-local-test",
+				ifaces:   map[string]string{},
+				hostConfig: &config.Config{
+					L3afConfigStoreFileName: "/etc/l3afd/l3af-config.json",
+				},
+			},
+			arg: []models.L3afBPFPrograms{
+				{
+					HostName: "enp0s3",
+					BpfPrograms: &models.BPFPrograms{
+						XDPIngress: []*models.BPFProgram{},
+						TCIngress:  []*models.BPFProgram{},
+						TCEgress:   []*models.BPFProgram{},
+					},
+				},
+			},
+			wanterr: true,
+		},
+		{
+			name: "UnknownInterface",
+			field: fields{
+				hostName: "l3af-local-test",
+				ifaces:   map[string]string{},
+				hostConfig: &config.Config{
+					L3afConfigStoreFileName: "/etc/l3afd/l3af-config.json",
+				},
+			},
+			arg: []models.L3afBPFPrograms{
+				{
+					HostName: "l3af-local-test",
+					Iface:    "dummyinterface",
+					BpfPrograms: &models.BPFPrograms{
+						XDPIngress: []*models.BPFProgram{
+							&models.BPFProgram{
+								Name:              "ratelimiting",
+								SeqID:             1,
+								Artifact:          "l3af_ratelimiting.tar.gz",
+								MapName:           "/sys/fs/bpf/xdp_rl_ingress_next_prog",
```

Comment:
I believe PR #78 should really get merged first and then this line should change to just
```suggestion
								MapName:           "xdp_rl_ingress_next_prog",
```

Same with line 505 etc

Reply:
done